src/gennylib/src/PoolManager.cpp

```diff
@@ -53,9 +55,22 @@ auto createPool(const std::string& name,
 }  // namespace genny::v1
 
 
-mongocxx::pool::entry genny::v1::PoolManager::client(const std::string& name,
-                                                     size_t instance,
-                                                     const Node& context) {
+mongocxx::pool::entry genny::v1::PoolManager::create_client(const std::string& name,
+                                                            size_t instance,
+                                                            const Node& context) {
+    auto pool_entry = this->_create_client(name, instance, context);
+    bool prewarm = context["Clients"][name]["PreWarm"].maybe<bool>().value_or(true);
+    if (prewarm && !_dryRun) {
+        auto ping = bsoncxx::builder::stream::document{} << "ping" << 1 << bsoncxx::builder::stream::finalize;
+        pool_entry->database("admin").run_command(ping.view());
+    }
+    return pool_entry;
+}
+
+
+mongocxx::pool::entry genny::v1::PoolManager::_create_client(const std::string& name,
```

Comment:
I initially wrap `create_client` around `_create_client` because of a conditional block for apm that returns the pool connection in seperate conditional branches. But after moving the pre-warming logic to its own helper function, this wrapping is no longer necessary.